indigo/indigo-extras/src/main/scala/indigoextras/waypoints/WaypointPath.scala

```diff
@@ -0,0 +1,113 @@
+package indigoextras.waypoints
+
+import indigo.Radians
+import indigo.Vector2
+
+import scala.annotation.tailrec
+
+trait WaypointPath:
+  def waypoints: List[Vector2]
+
+  def effectiveWaypoints: List[Vector2]
+
+  def loop: Boolean
+
+  def calculatePosition(
+      at: Double
+  ): (Vector2, Radians)
+
+object WaypointPath:
+  def apply(
+      waypoints: List[Vector2],
+      radius: Double,
```

Comment:
Is radius and expression of `Radians`, or is that too much?

Reply:
I'm not sure what do you mean by this? Radius is about the "proximty" to the waypoint that the actor needs to be from the waypoint for it to be considered visited. It's a measure of distance, not angle.